module/zfs/dnode.c

```diff
@@ -1778,15 +1778,24 @@ dnode_try_claim(objset_t *os, uint64_t object, int slots)
 }
 
 /*
- * Checks if the dnode contains any uncommitted dirty records.
+ * Checks if the dnode itself is dirty, or is carrying any uncommitted records.
+ * Its important to check both, as some operations (eg appending to a file) can
+ * dirty both as a single logical unit, but they are not synced out atomically,
+ * so checking one and not the other can result an object appearing to be clean
```

Comment:
can result an object appearing to be clean --->
can result in an object appearing to be clean

Reply:
I will do so, sorry for the inconvenience.